### ISSUE TITLE
Translation: we need to translate the code

### ISSUE THREAD

**Issue by Charlotte-Morgan** [inasafe/inasafe#3678]
# problem
InaSAFE v 4.0 has many new strings to be translated

# proposed solution
@felix-yew will help to translate short strings with no tags.
@ivanbusthomi will help to translate more complicated english terms and strings with tags

**Comment by timlinux** [comment]
Thanks very much! @ismailsunni to provide support if they need any help with the process...

**Comment by ismailsunni** [comment]
Sure, I will help :)

**Comment by ismailsunni** [comment]
Some comments:

English text | Your translation |  Proper translation | Notes
----------- | ---------------- | ------------------ | ------
from {number_affected} affected | dari {jumlah_terdampak} terdampak | dari {number_affected} terdampak | Do not translate sting inside { } 
Analysis initialisation | Analisis Inisialisasi | Inisialisasi Analisis | 
Hazard preparation | Persiapan ancaman bencana | Persiapan ancaman | hazard = ancaman
Aggregate hazard |  | ancaman teragregasi | 
Global Default | Default Umum | Default Global | 

I have fixed the one that I reviewed.

I will add more cc @felix-yew @ivanbusthomi

**Comment by ismailsunni** [comment]
Propossed date for the translation:
- [ ] 9 Feb 2017 : All strings are translated
- [ ] 10 - 13 Feb 2017 : Additional urgent strings / translation
- [ ] 14 Feb 2017 : Merging translation to develop branch

cc @timlinux @Charlotte-Morgan @ivanbusthomi @felix-yew @Gustry

**Comment by Gustry** [comment]
> hazard = ancaman

@ismailsunni You should use the glossary for that : 
https://www.transifex.com/inasafe/inasafe-code/glossary/fr/

Sometimes, one english word can have many equivalents in French. Using the glossary, I put one word in French so I always use the same word later. I also put notes sometimes.

**Comment by ismailsunni** [comment]
@Gustry ah thanks, I forget that transifex has it. I will check

Btw, I just push latest strings to transifex. Hopefully there won't be much change later.

**Comment by Gustry** [comment]
> Btw, I just push latest strings to transifex. Hopefully there won't be much change later.

We should have merge this PR : https://github.com/inasafe/inasafe/pull/3775

**Comment by ivanbusthomi** [comment]
Hi mas @ismailsunni ,
I check today there is no string to be translated for Indonesian, but some to be reviewed.
And the glossary feature in Transifex is a great help to standardize the translation of some words and terms.

**Comment by Charlotte-Morgan** [comment]
Hi @ivanbusthomi - sometimes i see some strings that look dodgy in English. When you are translating and see dodgy English, please let someone (me) know and we can attend to that before you translate it

**Comment by Gustry** [comment]
Didn't you get some issues like this in transifex ?
'Default value for feature rate per mÂ²' ?

**Comment by ismailsunni** [comment]
@ivanbusthomi yes, I have finished all translation on Tuesday night. But, new strings will come soon.
@Charlotte-Morgan yes, I also recognize some typos after working in transifex.
@Gustry I did. But I just copy and paste the `mÂ²`. Not sure if it will work or not.

Note to other developer @timlinux @myarjunar , I think it's a good practice to set the string in python code, not in .ui file. The reason is, there are html tag if we do in .ui file. It's prone error when we do translation.

**Comment by Gustry** [comment]
> Note to other developer @timlinux @myarjunar , I think it's a good practice to set the string in python code, not in .ui file. The reason is, there are html tag if we do in .ui file. It's prone error when we do translation.

I know you don't like having HTML in transifex, but for my side, translating in French, it's much easier to translate one sentence with HTML inside instead of having many small sentences in transifex. The order of words is slightly different. The perfect solution doesn't exist.

> I also recognize some typos after working in transifex.

There were there to fix them ;-)

**Comment by timlinux** [comment]
@Gustry OK sure happy to do things in the way that is easiest for translations...

**Comment by ismailsunni** [comment]
> I know you don't like having HTML in transifex, but for my side, translating in French, it's much easier to translate one sentence with HTML inside instead of having many small sentences in transifex. The order of words is slightly different. The perfect solution doesn't exist.

Yes, I agree for the part that you mentioned. But I was pointing at html boiler plate tag, like this
```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">⏎
<html><head><meta name="qrichtext" content="1" /><style type="text/css">⏎
p, li { white-space: pre-wrap; }⏎
</style></head><body style=" font-family:'AvantGarde Bk BT'; font-size:12pt; font-weight:400; font-style:normal;">⏎
<p style=" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;"><span style=" font-family:'Helvetica Neue,Helvetica,Segoe UI,Arial,freesans,sans-serif'; font-size:14px; color:#333333; background-color:#ffffff;">InaSAFE is free software that provides a simple but rigorous way to combine data from communities, local governments and scientists to help assess the possible impact of future disaster events on communities, assets and infrastructure. It uses innovative science and crowd sourcing techniques (OpenStreetMap, OSM) to create realistic disaster scenarios for better planning, preparedness and response activities. It is in demand by disaster managers across Indonesia and around the world. </span></p>⏎
<p style=" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;"><span style=" font-family:'Helvetica Neue,Helvetica,Segoe UI,Arial,freesans,sans-serif'; font-size:14px; color:#333333; background-color:#ffffff;">InaSAFE was developed jointly by the Indonesia's National Disaster Management Agency (BNPB), the Australian Government and the World Bank - Global Facility for Disaster Reduction and Recovery (World Bank-GFDRR).</span></p></body></html>
```
or this
```
<html><head/><body><p>Hazard<br/>classes</p></body></html>
```

**Comment by Gustry** [comment]
Some comments after @ismailsunni https://github.com/inasafe/inasafe/issues/3678#issuecomment-277179419
> 'In the event of a [hazard], how many [exposure] might be [impacted]?'

We should translate between square brackets.

**Comment by ismailsunni** [comment]
Note for developers @myarjunar @Gustry @timlinux @lucernae 
If you want to use `tr` function and token or argument, please make sure the `tr` method only get the raw string as parameter. Perhaps with example is easier:
```python
test_1 = tr('test 1 hazard name %s' % hazard['name'])
test_2 = tr('test 2 hazard name %s') % hazard['name']
test_3 = tr('test 3 hazard name {hazard_name}').format(hazard_name=hazard['name'])
test_4 = tr('test 4 hazard name {hazard_name}'.format(hazard_name=hazard['name']))
```

test 1 and test 4, will not work
test 2 and test 3 will work properly

It's better to write like test 3, easier for the translator.

**Comment by lucernae** [comment]
:+1: for test_3, I'm recommending this too since translator can reorder the words if necessary. with test_2, they can't do that.

**Comment by Gustry** [comment]
Indonesian is 100%, I'm removing the release blocker from this issue.
I'm still working on the French translations but it's not a blocker as we have 96%.

**Comment by ismailsunni** [comment]
Yes, death to `%s`.
Another note, please don't put `tr` in your logger. Most user won't see it and make the translator happier.

**Comment by Gustry** [comment]
Can you summarize these rules @ismailsunni in the wiki ?
Because this ticket will be closed soon. Thanks

**Comment by Gustry** [comment]
Another one I found : 
```
    'help_text': tr(
        '"Youth" is defined as: ' + concepts['youth']['description'] +
        'In cases where population data is available, InaSAFE will calculate '
        'the number of youths per exposure feature, aggregate hazard area, '
        'aggregation area and for the analysis area as a whole. The youth '
        'count is calculated based on standard ratios either provided as a '
        'global setting in InaSAFE, or (if available) counts or ratios in the '
        'input analysis data.'),
```
In transifex, we only see : 
`"Youth" is defined as:`
so we missed everything after a concatenation. We should not do this.

**Comment by ismailsunni** [comment]
@gustry Thanks 

Will add it to the wiki.
